File: plotProbSAT.R
#!/usr/bin/env Rscript
args = commandArgs(trailingOnly = TRUE)
file = args[1]
nPoints = 30
startValue = 1.0
endValue = 20.0
#load .csv file
file.path("statistics", paste(file, ".csv", sep=""))
experiments = read.csv(file.path("statistics", paste(file, ".csv", sep="")), 
                       header = FALSE, sep = ",", row.names = 1)
# setup x values
x = matrix(seq(startValue, endValue, length.out = nPoints), ncol = 1)
#setup output file
png(filename=file.path("plots/", paste(file, ".png", sep="")))
#plot
col_set = rainbow(nrow(experiments))
matplot(x, t(experiments), type = "l", lty = "solid", col= col_set, ylab = "non Horn clauses", xlab = "cb")
legend("right", rownames(experiments), lwd=c(2.5), col = col_set)
#close output file
dev.off()

File: plotProbSAT.m
#!/usr/bin/octave -f
arg_list = argv();
file_name = arg_list{1};
cmStart = str2double(arg_list{2});
cmEnd = str2double(arg_list{3});
cbStart = str2double(arg_list{4});
cbEnd = str2double(arg_list{5});
nPoints  = str2num(arg_list{6});
m = csvread(strcat('statistics/', file_name, '.csv'));
x = linspace(cmStart, cmEnd, nPoints);
y = linspace(cbStart, cbEnd, nPoints);
imagesc(x, y, m);
set(gca,'YDir','normal');
xlabel('cm');
ylabel('cb');
title('Number of non Horn clauses');
colormap(gray);
colorbar;
print(strcat('plots/', file_name, '.png'), '-dpng');